src/Misc/errorCodes.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Error codes for babylon JS errors
+ */
+export const ErrorCodes = Object.freeze({
+    // Mesh validation
+    MeshInvalidPositionsError: "MeshInvalidPositionsError",
+
+    // Material validation
+    UnsupportedTextureError: "UnsupportedTextureError",
+
+    // GLTF Loader errors
+    GLTFLoaderUnexpectedMagicError: "GLTFLoaderUnexpectedMagicError",
+
+    // Scene Load errors
+    SceneLoaderError: "SceneLoaderError",
+});
```

Comment:
Why use strings for error codes?

Reply:
Either is fine with me. String error codes are more flexible and explanatory so I used it. But if there is a best practice we should be following here I can make the change here. Or if there is any preferred way in open source projects for user-facing error codes.

Reply:
I don't know if there is a best practice. @RaananW @sebavan @deltakosh any thoughts?

Reply:
I would rather use a const enum, which is technically almost the same solution. I don't see a reason to use strings, as the variable name is of more important when handling these errors. You can also use a "as const" object:

```
const ErrorCodes = {
  FirstError: "FirstError",
  SecondError: "SecondError"
} as const
```
which is typescript's recommendation when complex enum features are not needed - https://www.typescriptlang.org/docs/handbook/enums.html#objects-vs-enums